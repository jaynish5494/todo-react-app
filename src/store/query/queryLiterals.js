export const queryLiterals = name => ({
  REQUEST: `${name}_REQUEST`,
  SUCCESS: `${name}_SUCCESS`,
  ERROR: `${name}_ERROR`
});

export const actionCreator = type => values => ({
  type,
  ...values
});

export const queryActions = name => {
  if (typeof name !== 'string' && !(name instanceof String)) {
    throw new Error('query action creators expect string for initialisation');
  }

  const { REQUEST, SUCCESS, ERROR } = queryLiterals(name);

  return {
    request: actionCreator(REQUEST),
    success: actionCreator(SUCCESS),
    error: actionCreator(ERROR)
  };
};
