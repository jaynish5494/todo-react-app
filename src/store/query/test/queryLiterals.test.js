import { queryLiterals, queryActions, actionCreator } from '../queryLiterals';

const name = 'TEST';

describe('query Actions', () => {
  it('should throw an error if called without a string argument', () => {
    expect(() => {
      queryActions();
    }).toThrow();
  });

  it('should return proper query action creators if initialised correctly', () => {
    const mockQueryLiterals = jest.fn();
    mockQueryLiterals.mockImplementation(() => queryLiterals(name));
    const { REQUEST, SUCCESS, ERROR } = mockQueryLiterals();

    const mockActionCreator = jest.fn();
    mockActionCreator.mockImplementation(actionCreator);

    const values = {
      value1: 'value1',
      value2: 1,
      value3: true
    };

    const expected = {
      request: mockActionCreator(REQUEST),
      success: mockActionCreator(SUCCESS),
      error: mockActionCreator(ERROR)
    };

    const actual = queryActions('TEST');
    expect(actual.request(values)).toEqual(expected.request(values));
    expect(actual.success(values)).toEqual(expected.success(values));
    expect(actual.error(values)).toEqual(expected.error(values));
  });
});

describe('query literals creator', () => {
  it('should create correct literals', () => {
    expect(queryLiterals(name)).toMatchSnapshot();
  });
});
