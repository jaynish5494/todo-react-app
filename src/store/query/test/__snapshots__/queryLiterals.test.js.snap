// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query literals creator should create correct literals 1`] = `
Object {
  "ERROR": "TEST_ERROR",
  "REQUEST": "TEST_REQUEST",
  "SUCCESS": "TEST_SUCCESS",
}
`;
